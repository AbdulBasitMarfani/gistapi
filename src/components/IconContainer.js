import styled from "styled-components";
import Octicon from "react-octicon";

const IconContainer = ({ iconName, link, text }) => {
  return (
    <Wrapper>
      <Octicon name={iconName} />
      {link ? (
        <AnchorWrapper href={link} target="_blank">
          {text}
        </AnchorWrapper>
      ) : (
        <TextWrapper>{text}</TextWrapper>
      )}
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: inline-flex;
  gap: 0.25rem;
  justify-content: center;
  align-items: center;
//   width: 100px;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--primary-color);
`;

const AnchorWrapper = styled.a`
  text-decoration: none;
  font-size: 12px;
  color: var(--primary-color);
`;

const TextWrapper = styled.span`
  font-size: 12px;
`;

export default IconContainer;
