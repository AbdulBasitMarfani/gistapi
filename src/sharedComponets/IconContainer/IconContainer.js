import styled from "styled-components";
import Octicon from "react-octicon";

export const IconContainer = ({ iconName, link, text }) => {
  return (
    <Wrapper>
      <Octicon name={iconName} />
      {link ? (
        <AnchorWrapper href={link} target="_blank">
          {text}
        </AnchorWrapper>
      ) : (
        <TextWrapper>{text}</TextWrapper>
      )}
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: inline-flex;
  gap: 0.25rem;
  justify-content: center;
  align-items: center;
  text-overflow: ellipsis;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  color: var(--primary-color);
`;

const AnchorWrapper = styled.a`
  text-decoration: none;
  font-size: 12px;
  color: var(--primary-color);
`;

const TextWrapper = styled.span`
  font-size: 12px;
`;
