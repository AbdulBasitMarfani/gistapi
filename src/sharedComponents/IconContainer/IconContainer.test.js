import { render } from "@testing-library/react";
import { IconContainer } from "./IconContainer";

const props = { iconName: "code", text: "Files" };

describe("Gist", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<IconContainer {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
