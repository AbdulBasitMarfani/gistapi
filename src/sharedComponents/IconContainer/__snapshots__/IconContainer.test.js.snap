// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gist should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-aXZVg iEJzdC"
  >
    <span
      class="octicon octicon-code"
    />
    <span
      class="sc-eqUAAy cYzGHt"
    >
      Files
    </span>
  </div>
</DocumentFragment>
`;
